eos_data_distribution/avahi_monitor.py

```diff
@@ -188,6 +192,12 @@ def remove_nexthop(self, faceUri):
         return check_call(["nfdc", "remove-nexthop",
                            str(SUBSCRIPTIONS_SOMA), faceUri])
 
+    def start_edge(self):
+        return check_call(["systemctl", "start", "edd-soma-subscriptions-producer"])
```

Comment:
I don’t think this is the right architecture. This effectively means that the Avahi service runs to work out whether the `edd-soma-subscriptions-producer` can run — why not move the Avahi code into `edd-soma-subscriptions-producer`, so that the state information is kept within process, rather than being communicated between two processes via `systemctl` calls?